src/Mod/Mesh/App/Exporter.h

```diff
@@ -42,42 +43,25 @@ namespace Mesh
  * Constructors of derived classes are expected to be required, for passing
  * in the name of output file.
  *
- * Objects are added using the addMeshFeat(), addPartFeat(), etc.
- *
  * If objects are meant to be combined into a single file, then the file should
  * be saved from the derived class' destructor.
  */
 class Exporter
 {
     public:
         Exporter();
+        virtual ~Exporter() = default;
 
-        /*!
-         * \return true if \a is an object that can be exported as mesh.
-         */
-        static bool isSupported(App::DocumentObject *obj);
-
-        virtual bool addMeshFeat(App::DocumentObject *obj) = 0;
-        virtual bool addPartFeat(App::DocumentObject *obj, float tol) = 0;
-
-        /// Recursively adds objects from App::Part & App::DocumentObjectGroup
-        /*!
-         * \return true if all applicable objects within the group were
-         * added successfully.
-         */
-        bool addAppGroup(App::DocumentObject *obj, float tol);
-
-        bool addObject(App::DocumentObject *obj, float tol);
+        int addObject(App::DocumentObject *obj, float tol);
 
-        virtual ~Exporter() = default;
+        virtual bool addMesh(const char *name, const MeshObject & mesh) = 0;
```

Comment:
Should this be part of the public interface of the base class?
As far as I can tell it is only called from `addObject`, which in turn is only called from a single location in `AppMeshPy.cpp`

Reply:
Although this is expected to be used by `addObject()` only, there is no harm for others to call this function to add mesh. That's why it is `public`. The `virtual` is because there are two exporter class, `MergeExporter` and `AmfExporter` which has different ways of storing the meshes.